bot/exts/moderation/infraction/infractions.py

```diff
@@ -480,6 +474,47 @@ async def apply_ban(
         async def action() -> None:
             await ctx.guild.ban(user, reason=reason, delete_message_days=purge_days)
 
+        if isinstance(user, Member) and any(role.id in constants.STAFF_PARTNERS_COMMUNITY_ROLES for role in user.roles):
+            confirmation_view = _utils.StaffBanConfirmationView(
+                allowed_users=(ctx.author.id,),
+                allowed_roles=constants.MODERATION_ROLES,
+                timeout=10,
+            )
+
+            await confirmation_view.send(
+                ctx.channel,
+                message_content=f"{user} has an elevated role. Are you sure you want to ban them?",
+            )
```

Comment:
```suggestion
            message = await ctx.send(
                "{user} has an elevated role. Are you sure you want to ban them?",
                view=confirmation_view
            )
            confirmation_view.message = message
```

This way would be consistent with howt he view is used in snekbox, and removes the need for both the init and the send functions on the subclass StaffBanConfirmationView